### PR TITLE
Bootstrap 5.3

### PR DESCRIPTION
**Pull request by skyporter**
- bundle update gem `boostrap` and `bootstrap_form` 
- move some model translations into the right location
- update forms according to the new BS5 form structuration (ie. `form-group`=> `mb-3`) + add missing form classes (ie. `form-label`)
- update BS5 JS selectors of dropdown, modal, popover (ie. `data-toggle="dropdown"`=> `data-bs-toggle="dropdown"`)
- update to jquery3 which is a requirement of BS3
- update css classes to the new BS5 selectors (`ml-x`=> `ms-X`, `dropdown-menu-right` => `dropdown-menu-end`, ...)
- multiple design updates to look and feel like the current application
- add missing `hr` on forms before the primary submit button
- update `bootstrap4-datetimepicker` to `tempus-dominus`

### CODE REVIEW

**Comment by lucaguindani** [comment]
@skyporter you can find below the issues I found so far.

- [ ] [advanced-search-config](https://github.com/catima/catima/assets/1675064/30c45987-afce-46bc-8ee5-8411b971fe91)
- [ ] [categories-not-hidden](https://github.com/catima/catima/assets/1675064/362a2a3b-d769-4fc0-9ee1-b610d5db042d)
- [ ] [footnotes-html-tags](https://github.com/catima/catima/assets/1675064/bc92935f-2f4e-44e6-8bd2-3fd6c499b82c)
- [ ] [html-pages-summernote](https://github.com/catima/catima/assets/1675064/9c4da18a-be9c-4f2a-b120-0d73b89db29e)
- [ ] [item-last-field-hr](https://github.com/catima/catima/assets/1675064/9090b1e7-6849-4f57-ab44-c4794bdd661b)
- [ ] [item-list-overflow](https://github.com/catima/catima/assets/1675064/d14e77dd-8a4c-409f-b135-5fc475cb672a)
- [ ] [map-container-items](https://github.com/catima/catima/assets/1675064/c8ceafa3-fc95-421d-ad6e-1bc5c065ecd7)
- [ ] [multilingual-formatted-text](https://github.com/catima/catima/assets/1675064/0bf59863-f289-4b92-9a10-7e6f6c4f9fb9)
- [ ] [readonly-content-should-be-grayed](https://github.com/catima/catima/assets/1675064/8facdfc3-e513-4a2d-ac29-e4b78eca38c2)

File: app/controllers/catalog_admin/csv_imports_controller.rb [hunk 1]
```diff
@@ -38,6 +38,7 @@ def build_csv_import(params=nil)
     @csv_import = CSVImport.new(params) do |import|
       import.creator = current_user
       import.item_type = item_type
+      import.file_encoding = "detect"
```
Comment (lucaguindani) [inline app/controllers/catalog_admin/csv_imports_controller.rb:41]: Why ?

File: package.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
     "@babel/preset-react": "^7.12.5",
     "@babel/preset-typescript": "^7.12.1",
     "@babel/runtime": "^7.16.7",
+    "@eonasdan/tempus-dominus": "^6.9.6",
+    "@popperjs/core": "^2.11.8",
```
[the remark this answers is not in the document]
Comment (skyporter) [reply package.json:12]: c'est une dépendance obligatoire du nouveau datepicker tempus-dominus

File: app/controllers/catalog_admin/csv_imports_controller.rb (hunk 1 above)
Comment (skyporter) [reply app/controllers/catalog_admin/csv_imports_controller.rb:41]: sinon les tests plantent

Comment (skyporter) [reply app/controllers/catalog_admin/csv_imports_controller.rb:41]: screenshot 2024-05-14 at 11 16 58

en haut sans le `import.file_encoding = "detect"`, en bas avec.

**Comment by skyporter** [comment]
@lucaguindani

1. _footnotes-html-tags_ => [by default tootlip](https://getbootstrap.com/docs/5.2/components/popovers/#options) are now html safe (prevent XSS)... I'll change that.
2.  _item-last-field-hr_ => this due by the new table formatting, now tables add border at the bottom and not at the top... I'll remove the additional hr.
3.  _readonly-content-should-be-grayed_ => that's what I told you, https://getbootstrap.com/docs/5.2/forms/form-control/#readonly is ‘white’ by default. Should I override the default CSS?

**Comment by lucaguindani** [comment]
> 1. _footnotes-html-tags_ => [by default tootlip](https://getbootstrap.com/docs/5.2/components/popovers/#options) are now html safe (prevent XSS)... I'll change that.

✅

> 2. _item-last-field-hr_ => this due by the new table formatting, now tables add border at the bottom and not at the top... I'll remove the additional hr.

✅

> 3. _readonly-content-should-be-grayed_ => that's what I told you, https://getbootstrap.com/docs/5.2/forms/form-control/#readonly is ‘white’ by default. Should I override the default CSS?

Yes please

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucaguindani) [reply package.json:12]: ✅

File: app/controllers/catalog_admin/csv_imports_controller.rb (hunk 1 above)
Comment (lucaguindani) [reply app/controllers/catalog_admin/csv_imports_controller.rb:41]: Strange... I don't see what bootstrap or jquery have to do with it.

**Comment by skyporter** [comment]
2. _item-last-field-hr_ => not possible to remove the "hr" because defined on footer. I'm looking for removing border on last table tr.

**Comment by lucaguindani** [comment]
> * [ ]  [html-pages-summernote](https://github.com/catima/catima/assets/1675064/9c4da18a-be9c-4f2a-b120-0d73b89db29e)

@skyporter seems summernote has a bs5 version available https://cdnjs.com/libraries/summernote. But I don't know if the `summernote-rails` gem is compatible (not updated since 2019).

https://github.com/catima/catima/blob/development/app/assets/stylesheets/application.scss#L5
https://github.com/catima/catima/blob/development/app/assets/javascripts/catalog-admin-core.js#L3

**Comment by skyporter** [comment]
All issues except [html-pages-summernote](https://github.com/catima/catima/assets/1675064/9c4da18a-be9c-4f2a-b120-0d73b89db29e) are fixed.

About this last issue with summernote, it works for "compount" components which only contains custom dropdown. I forgot to fix summernote on "html containers" where the dropdown is in conflicts with the tooltip.

I tried summernote-b5 and summernote-lite but got errors.

**Comment by skyporter** [comment]
summernote code has been updated https://github.com/summernote/summernote/commit/702f09a4f0675eadeab87cf9760794b633edb5ac but I can't found a package with this new code. 
I think we're going to have to compile this javascript ourselves.

**Comment by skyporter** [comment]
screenshot 2024-05-14 at 19 09 57

One solution would be to switch to summernote lite, but you're going to have to do some css because it's all broken.

**Comment by skyporter** [comment]
I managed to get summernote to work with the lite version from CDN.

I suggest removing the gem and adding the yarn package. It works. However, there's still a problem with the package, the CSS can't load the fonts... simple solution: use the CDN to load the CSS, is that possible? 

With the JS pacakge from `node_modules` and CSS from CDN:
screenshot 2024-05-14 at 19 20 57

With both JS and CSS from `node_modules`:
screenshot 2024-05-14 at 19 44 45

**Comment by lucaguindani** [comment]
> I managed to get summernote to work with the lite version from CDN.
> 
> I suggest removing the gem and adding the yarn package. It works. However, there's still a problem with the package, the CSS can't load the fonts... simple solution: use the CDN to load the CSS, is that possible?

@skyporter, the CSS cannot be added in https://github.com/catima/catima/blob/development/app/assets/stylesheets/application.scss or https://github.com/catima/catima/blob/development/app/assets/stylesheets/catalog-admin.scss like we do for other js libs (quill, react-leaflet-markercluster) ?

**Comment by skyporter** [comment]
> > I managed to get summernote to work with the lite version from CDN.
> > I suggest removing the gem and adding the yarn package. It works. However, there's still a problem with the package, the CSS can't load the fonts... simple solution: use the CDN to load the CSS, is that possible?
> 
> @skyporter, the CSS cannot be added in https://github.com/catima/catima/blob/development/app/assets/stylesheets/application.scss or https://github.com/catima/catima/blob/development/app/assets/stylesheets/catalog-admin.scss like we do for other js libs (quill, react-leaflet-markercluster) ?

Yes for the CSS file, but there are fonts referenced in this css that don't load correctly (see screenshot in my previous comment).

**Comment by skyporter** [comment]
argh, sorry I click on close... rollbacked

**Comment by lucaguindani** [comment]
> Yes for the CSS file, but there are fonts referenced in this css that don't load correctly (see screenshot in my previous comment).

Loading the css from a CDN is not ideal, but if there is no other possibility let's do that. We'll have to remember to check the css version match when the library is updated via yarn.

**Comment by lucaguindani** [comment]
> * [ ]  [search-results](https://github.com/catima/catima/assets/1675064/ff28ec4b-bbb2-4c43-ae12-8919a14598d7) Missing item count badges, wrong text color in tabs if catalog as a custom style

@skyporter found a new one

**Comment by skyporter** [comment]
> > Yes for the CSS file, but there are fonts referenced in this css that don't load correctly (see screenshot in my previous comment).
> 
> @skyporter Loading the css from a CDN is not ideal, but if there is no other possibility let's do that. We'll have to remember to check the css version match when the library is updated via yarn.

I finally found a solution by overiding the src url for font from our scss:
```
@font-face {
  font-family: "summernote";
  src: url("summernote/src/font/summernote.eot?#iefix") format("embedded-opentype"), url("summernote/src/font/summernote.woff2") format("woff2"), url("summernote/src/font/summernote.woff") format("woff"), url("summernote/src/font/summernote.ttf") format("truetype");
}
```

No more need for CDN.

**Comment by skyporter** [comment]
> * [ ]   [spacing-search-results](https://github.com/catima/catima/assets/1675064/22c24353-ed04-49ba-bd79-ba3713968abd), [spacing-simple-search](https://github.com/catima/catima/assets/1675064/c3d91b80-7f0a-4af7-944e-492f82ceb2cb)

fixed

**Comment by skyporter** [comment]
> * [ ]  [html-pages-summernote](https://github.com/catima/catima/assets/1675064/9c4da18a-be9c-4f2a-b120-0d73b89db29e)

fixed

**Comment by skyporter** [comment]
> [ ] [search-results](https://github.com/catima/catima/assets/1675064/f0a8bbc9-c8ee-45ef-8d8a-509ab2039234), [advanced-search](https://github.com/catima/catima/assets/1675064/e34e2e62-7e8d-4bfc-8a0e-ca1657a1dd94) wrong links color if the catalog as a [custom style](https://github.com/catima/catima/assets/1675064/719f18b6-0f50-4d4b-84ef-2fe1b9379df4)

The current logic on BS4 with the colour of the links is incomprehensible. Sometimes the links are overloaded, sometimes not. For example, `nav-link` and `btn-link` don't have the same logic. And what about the links in the navigation, should they be overloaded or not? I though not, but what about nav-link on `nav-tabs`?

What's the ideal? 

I updated the code to get something that, I think, is consistent: links (a) + .btn-text and .btn-link + .nav-tabs .nav-link.

**Comment by skyporter** [comment]
> [ ] [tootip-behavior1](https://github.com/catima/catima/assets/1675064/e24ef1af-4804-45ad-89b0-63e0950f6fe7), [tootip-behavior2](https://github.com/catima/catima/assets/1675064/c5c89561-7701-4ab5-8bc2-fd225fc34f7a)

fixed

**Comment by lucaguindani** [comment]
> > * [ ]    [favorites-badges](https://github.com/catima/catima/assets/1675064/aa78e29d-e61a-4c73-b22d-c4ea645c5e46)
> 
> fixed

@skyporter found a new issue with favorites with thumbnails [favorites-with-thumbnails](https://github.com/catima/catima/assets/1675064/ebfa0434-4f9a-4fbe-b80c-a80ba4343ad5)

**Comment by lucaguindani** [comment]
> > * [ ]   [html-pages-summernote](https://github.com/catima/catima/assets/1675064/9c4da18a-be9c-4f2a-b120-0d73b89db29e)
> 
> fixed

@skyporter looks fine for the most part. Except for the following two issues [summernote-source-code](https://github.com/catima/catima/assets/1675064/f184a5e4-e67c-4f46-a87d-28691b815c59), [summernote-video-embed](https://github.com/catima/catima/assets/1675064/c3dfbf6f-6943-4bf2-82d9-50ec140033a6)

**Comment by lucaguindani** [comment]
> * [ ]  [list-with-thumbnails](https://github.com/catima/catima/assets/1675064/bcade5a6-d0a7-44e4-b6f0-d52e8ad452a4)

@skyporter found a new issue in items index with list style & thumbnails. Looks like something similar to the [favorites-badges](https://github.com/catima/catima/assets/1675064/aa78e29d-e61a-4c73-b22d-c4ea645c5e46) issue.

**Comment by lucaguindani** [comment]
> > [ ] [search-results](https://github.com/catima/catima/assets/1675064/f0a8bbc9-c8ee-45ef-8d8a-509ab2039234), [advanced-search](https://github.com/catima/catima/assets/1675064/e34e2e62-7e8d-4bfc-8a0e-ca1657a1dd94) wrong links color if the catalog as a [custom style](https://github.com/catima/catima/assets/1675064/719f18b6-0f50-4d4b-84ef-2fe1b9379df4)
> 
> The current logic on BS4 with the colour of the links is incomprehensible. Sometimes the links are overloaded, sometimes not. For example, `nav-link` and `btn-link` don't have the same logic. And what about the links in the navigation, should they be overloaded or not? I though not, but what about nav-link on `nav-tabs`?
> 
> What's the ideal?
> 
> I updated the code to get something that, I think, is consistent: links (a) + .btn-text and .btn-link + .nav-tabs .nav-link.

Looks good to me !

**Comment by skyporter** [comment]
About [summernote-video-embed](https://github.com/catima/catima/assets/1675064/c3dfbf6f-6943-4bf2-82d9-50ec140033a6), I have the same issue on BS4
screenshot 2024-05-17 at 16 10 23.

I think this file https://github.com/digITpro/catima/blob/c5b177a0223daed274059f5e15cee248ce47c087/config/initializers/secure_headers.rb#L26 should be updated to allow this origin.

**Comment by lucaguindani** [comment]
> About [summernote-video-embed](https://github.com/catima/catima/assets/1675064/c3dfbf6f-6943-4bf2-82d9-50ec140033a6), I have the same issue on BS4
> I think this file https://github.com/digITpro/catima/blob/c5b177a0223daed274059f5e15cee248ce47c087/config/initializers/secure_headers.rb#L26 should be updated to allow this origin.

Yes you're right, it should work on staging/production with https.

**Comment by skyporter** [comment]
about [summernote-source-code](https://github.com/catima/catima/assets/1675064/f184a5e4-e67c-4f46-a87d-28691b815c59), there's html `<style>` meta code on content. The problem comes from this `<style>` which removed by summernote. I've checked the possible parameters and found nothing to fix that.

screenshot 2024-05-17 at 16 20 21

One possible solution should be to parse content before rendering in summernote, to automatically remove all `<style>` tags rom content.

**Comment by lucaguindani** [comment]
> about [summernote-source-code](https://github.com/catima/catima/assets/1675064/f184a5e4-e67c-4f46-a87d-28691b815c59), there's html `<style>` meta code on content. The problem comes from this `<style>` which removed by summernote. I've checked the possible parameters and found nothing to fix that.
> 
> One possible solution should be to parse content before rendering in summernote, to automatically remove all `<style>` tags rom content.

I think this was already the case before this PR. But since this PR the @font-face data appearing in the editor when we leave the source code mode ([summernote-source-code](https://github.com/catima/catima/assets/1675064/f184a5e4-e67c-4f46-a87d-28691b815c59)), I don't know why.

**Comment by skyporter** [comment]
> I think this was already the case before this PR. But since this PR the @font-face data appearing in the editor when we leave the source code mode ([summernote-source-code](https://github.com/catima/catima/assets/1675064/f184a5e4-e67c-4f46-a87d-28691b815c59)), I don't know why.

because the editor deletes the `<style>` tag without deleting its content (which contains lots of style definitions).

**Comment by skyporter** [comment]
> > > * [ ]    [list-with-thumbnails](https://github.com/catima/catima/assets/1675064/bcade5a6-d0a7-44e4-b6f0-d52e8ad452a4)
> > 
> > 
> > @skyporter found a new issue in items index with list style & thumbnails. Looks like something similar to the [favorites-with-thumbnails](https://github.com/catima/catima/assets/1675064/ebfa0434-4f9a-4fbe-b80c-a80ba4343ad5) issue.
> 
> I didn't see this issue because I don't have the medias...

fixed

**Pull request merged**